source/dofs/dof_handler.cc

```diff
@@ -2453,13 +2441,40 @@ DoFHandler<dim, spacedim>::set_fe(const hp::FECollection<dim, spacedim> &ff)
 
   // don't create a new object if the one we have is already appropriate
   if (this->fe_collection != ff)
-    this->fe_collection = hp::FECollection<dim, spacedim>(ff);
+    {
+      this->fe_collection = hp::FECollection<dim, spacedim>(ff);
+
+      const bool contains_multiple_fes = (this->fe_collection.size() > 1);
+
+      // toggle the hp-mode of this object only if the type of the registered fe
+      // collection changed
+      if (hp_capability_enabled != contains_multiple_fes)
```

Comment:
One question. These lines suggest that we could toggle back from normal to hp mode. But that does not make sense, does it? We already have cleared the tables so that active fe indices couldn't have been set.

Reply:
Hmm, initially I thought that switching back from non-hp to hp mode would be possible. But you are correct: setting active and future fe indices is no longer possible once we've been in non-hp mode.

I will change this behavior and only allow the transition from hp to non-hp mode. If multiple fes are connected to non-hp mode, I will trigger an `AssertThrow`, so that users will be noticed also in RELEASE mode.